1-Draft/RFC00XX-PredIctive-IntelliSense.md

```diff
@@ -0,0 +1,359 @@
+Users may search the entire prediction history for commands matching under the current cursor
+position. Users may search prediction history based on 'Verb', 'Noun' and 'Parameter' of a cmdlet
+using the up/down arrow keys.
+
+* Searching history from cmdlet verb.
+![Verb](./media/pi-arrow-verb.gif)
+
+* Searching histroy from cmdlet noun.
+![noun](./media/pi-arrow-noun.gif)
+
+* Searching history from parameter.
+![Param](./media/pi-arrow-param.gif)
+
+### Edit predictions before accepting the prediction
+
+Users may navigate prediction tokens to make changes to parameters, arguments, and additional
+pipeline commands. The prediction model will update the displayed prediction from the closest
+matches in PSReadLine history or added provider.
+
+In the example below, a user navigates the prediction, pausing to change the -Property argument and
+receiving an updated prediction.
+
+![arg change](./media/pi-fb.gif)
+
+## Prediction Extension Model
+
+Additional providers enhance historic predictions by providing domain specific commands and task
+completions. Predictive IntelliSense includes an extension model to support the registration of
+additional providers.
+
+## Goals/Non-Goals
+
+ Goals:
+
+* Provide extension model to support additional providers
+* Support the discovery of providers using Get-Subsystem
+* Render predictions from providers within performance guidelines 20ms
+* Provide Module owners method of registration
+
+Non-goals:
+
+* Register/Unregister-Subsystem cmdlets are not planned because it's targeting binary subsystem
+  implementations, which should deal with registration/unregistration via APIs.
+* Server-side provider privacy is the responsibility of the provider module
+
+## Extension Model Specification
+
+### Predictive Suggestion Contract to Support Addition Providers
+
+Below is an abbreviated code listing of the API contract to support suggestions from additional
+providers.
+
+```csharp
+/// The class represents a predictive suggestion generated by a predictor.
+public sealed class PredictiveSuggestion
+{
+/// Gets the suggestion.
+public string SuggestionText { get; }
+
+/// Gets the tooltip of the suggestion.
+public string? ToolTip { get; }
+
+/// Initializes a new instance of the <see cref="PredictiveSuggestion"/> class.
+/// <param name="suggestion">The predictive suggestion text.</param>
+public PredictiveSuggestion(string suggestion)
+: this(suggestion, toolTip: null)
+```
+
+### Discovery of Provider Metadata and Registration
+
+Providers are registered automatically when the user installs a provider module. Removal of
+registration occurs when the module is removed. Users will need to discover installed providers
+along with additional metadata, such as version information to assist users in upgrading and
+diagnosing problems.
+
+To list the currently installed providers and their respective metadata:
+
+```powershell
+Get-SubSystem
```

Comment:
Show sample output (what the table will look like)

Reply:
You need to use `Get-Subsystem -Kind CommandPredictor` to get the registered prediction plugins.
Note that the `CommandPredictor` is just one subsystem.

Reply:
Added sample output with example Get-Subsystem -Kind CommandPredictor